examples/arithmetics/src/language-server/arithmetics-validator.ts

```diff
@@ -73,6 +72,51 @@ export class ArithmeticsValidator {
         }
     }
 
+    checkFunctionRecursion(module: Module, accept: ValidationAcceptor): void {
+        const traversedFunctions = new Set<Definition>();
+        function* getNotTraversedNestedCalls(func: Definition): Generator<NestedFunctionCall> {
+            if (!traversedFunctions.has(func)) {
+                traversedFunctions.add(func);
+                yield* NestedFunctionCall.selectCalls(func);
+            }
+        }
+
+        const callsTree: FunctionCallTree = new Map<ResolvedFunctionCall, NestedFunctionCall>();
+        const callCycles: FunctionCallCycle[] = [];
+        const bfsStep = (parent: NestedFunctionCall): NestedFunctionCall[] => {
+            const referencedFunc = parent.call.func.ref;
+            const uncycledChildren: NestedFunctionCall[] = [];
+            if (parent.host === referencedFunc) callCycles.push([parent]);
```

Comment:
I would always write a `{ ... }` block in control structures for consistency and readability.